src/utils/formatNumber.ts

```diff
@@ -193,3 +193,22 @@ export const formatCurrency = (number: string | number, currency: string): strin
 
   return format(number, currencyFormatter, minimum)
 }
+
+export const localeNumberFormatter = (amount: number | string): string => {
+  // get locale decimal separator
+  const number = 123456.789
+  const localeNumber = number.toLocaleString()
+  // Find localeNumber thousand and decimal separators
+  const [thousandSeparator, decimalSeparator] = localeNumber.match(/[^0-9]/g) || []
+
+  const [integerString, decimalString] = amount.toString().split(decimalSeparator)
+
+  const decimalFinal = decimalString ? `${decimalSeparator}${decimalString}` : ''
+
+  // trailing zeros
+  const decimalFinalWithoutTrailingZeros = decimalString ? decimalFinal.replace(/0+$/, '') : ''
+
+  const integerFinal = integerString.replace(/\B(?=(\d{3})+(?!\d))/g, thousandSeparator)
```

Comment:
Are thousand separators universal/set every three digits in all number formats?

Reply:
Definitely not: https://learn.microsoft.com/en-us/globalization/locale/number-formatting
Hindi – two digits, Japanese four.

Reply:
Good point here, the regex was very opinionated. I'm using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/toLocaleString now to format amount's integer part.